### PR TITLE
Utilize api/pkg/resources/machine.Machine()

### LINKED ISSUE #1472: e2e fix machine spec initialization
Currently machine spec taken directly from input, this is wrong, since Machine spec should be initialized on controller level

TODO: use `api/pkg/resources/machine.Machine()` for that

### PR DESCRIPTION
**Pull request by kron4eg**
Fix machine object initialization by reusing `api/pkg/resources/machine.Machine()`

**Which issue(s) this PR fixes**:
Fixes #1472 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/cmd/kubermatic-e2e/e2e_test_runner.go [hunk 1]
```diff
@@ -229,20 +236,59 @@ func (ctl *e2eTestRunner) nodesReadyCond(ctx context.Context) func() (bool, erro
...
+	dcsBuf, found := secret.Data["datacenters.yaml"]
+	if !found {
+		return provider.DatacenterMeta{}, errors.New("datacenters.yaml not found")
+	}
+
+	dcs := struct {
+		Datacenters map[string]provider.DatacenterMeta `yaml:"datacenters"`
+	}{}
+
+	dcsJSON, err := kubeyaml.ToJSON(dcsBuf)
```
Comment (mrIncompetent) [inline api/cmd/kubermatic-e2e/e2e_test_runner.go]: I don't think you need to convert it to json.
The structs have the yaml tag as this is a custom object we load from a file

Comment (kron4eg) [reply api/cmd/kubermatic-e2e/e2e_test_runner.go]: @mrIncompetent yeah, you're right. Fixed that.

**Review by mrIncompetent — DISMISSED** [review]

File: api/cmd/kubermatic-e2e/main.go [hunk 2]
```diff
@@ -48,8 +50,8 @@ func main() {
 	flag.BoolVar(&runOpts.GinkgoNoColor, "ginkgo-nocolor", false, "don't show colors")
 	flag.DurationVar(&runOpts.GinkgoTimeout, "ginkgo-timeout", 5400*time.Second, "ginkgo execution timeout")
 	flag.StringVar(&runOpts.Focus, "ginkgo-focus", `\[Conformance\]`, "tests focus")
-	flag.StringVar(&runOpts.Skip, "ginkgo-skip", `Flaky`, "skip those groups of tests")
-	flag.IntVar(&runOpts.Parallel, "ginkgo-parallel", 3, "parallelism of tests")
+	flag.StringVar(&runOpts.Skip, "ginkgo-skip", `Alpha|Kubectl|\[(Disruptive|Feature:[^\]]+|Flaky)\]`, "skip those groups of tests")
```
Comment (mrIncompetent) [inline api/cmd/kubermatic-e2e/main.go]: Why skipping all of those?
Is Sonobuoy skipping this as well?

Comment (kron4eg) [reply api/cmd/kubermatic-e2e/main.go]: yep: https://github.com/heptio/kube-conformance/blob/master/Dockerfile#L33

Comment (mrIncompetent) [reply api/cmd/kubermatic-e2e/main.go]: They mention that kubectl does not use the in-cluster config.
As Sonobouy runs inside a cluster it makes, sense.
But in our case we always specify the kubeconfig, so those should succeed?

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**